explainer/explainer_pipe_interface.py

```diff
@@ -0,0 +1,88 @@
+import os
+import glob
+
+from explainer.lecture_app_engine import AppEngine
+
+
+class Pipe:
+    """Interface representing a pipe."""
+
+    def _create(self):
+        """Create method for the pipe. Raises NotImplementedError."""
+        raise NotImplementedError
+
+    def receive(self):
+        """Receive method for the pipe. Raises NotImplementedError."""
+        raise NotImplementedError
+
+    def send(self, lecture):
+        """Send method for the pipe. Raises NotImplementedError."""
+        raise NotImplementedError
+
+    def pop(self):
+        """Pop method for the pipe. Raises NotImplementedError."""
+        raise NotImplementedError
+
+
+class FileSystemPipe(Pipe):
+    """Class representing a file system pipe."""
+
+    def __init__(self, lectures_folder, destination_folder):
+        """
+        Initialize the FileSystemPipe.
+
+        Args:
+            lectures_folder (str): Path to the folder containing lectures.
+            destination_folder (str): Path to the destination folder.
+
+        """
+        self.__front = None
+        self._lectures_folder = lectures_folder
+        self._destination_folder = destination_folder
+        self._create()
+
+    def _create(self):
+        """Create method for the file system pipe."""
+        if not os.path.exists(self._lectures_folder):
+            # Create the directory
+            os.makedirs(self._lectures_folder)
+        if not os.path.exists(self._destination_folder):
+            # Create the directory
+            os.makedirs(self._destination_folder)
+
+    def send(self, lecture):
+        """
+        Send the lecture through the file system pipe.
+
+        Args:
+            lecture: The lecture to send.
+
+        """
+        explainer_engine = AppEngine(lecture, self._destination_folder)
```

Comment:
you're creating an new object at each call for send ??

Reply:
.